flink-end-to-end-tests/run-nightly-tests.sh

```diff
@@ -182,5 +205,29 @@ run_test "Local recovery and sticky scheduling end-to-end test" "$END_TO_END_DIR
 run_test "Local recovery and sticky scheduling end-to-end test" "$END_TO_END_DIR/test-scripts/test_local_recovery_and_scheduling.sh 4 10 rocks true true" "skip_check_exceptions"
 
 printf "\n[PASS] All bash e2e-tests passed\n"
-printf "\n Please remember to run java e2e-tests by running mvn verify -Dcategories=\"\" -DdistDir=<path to flink-dist> in flink-end-to-end-tests\n"
+
+printf "\n\n==============================================================================\n"
+printf "Running Java end-to-end tests\n"
+printf "==============================================================================\n"
+
+HERE="`dirname \"$0\"`"				# relative
+HERE="`( cd \"${HERE}\" && pwd -P)`" 	# absolutized and normalized
```

Comment:
This resolves to `flink/flink-end-to-end-tests`, no? (based on the code at the top, which is identical).
If so, then you are excluding all walkthrough modules since we will not be found.

Reply:
It found the walkthrough modules?!

```
2020-02-28T03:53:27.2778157Z [INFO] flink-cli-test ..................................... SUCCESS [  6.257 s]
2020-02-28T03:53:27.2779191Z [INFO] flink-parent-child-classloading-test-program ....... SUCCESS [  0.374 s]
2020-02-28T03:53:27.2780199Z [INFO] flink-parent-child-classloading-test-lib-package ... SUCCESS [  0.235 s]
2020-02-28T03:53:27.2797292Z [INFO] flink-dataset-allround-test ........................ SUCCESS [  0.234 s]
2020-02-28T03:53:27.2798579Z [INFO] flink-dataset-fine-grained-recovery-test ........... SUCCESS [  1.567 s]
2020-02-28T03:53:27.2799589Z [INFO] flink-datastream-allround-test ..................... SUCCESS [  3.847 s]
2020-02-28T03:53:27.2800549Z [INFO] flink-batch-sql-test ............................... SUCCESS [  0.272 s]
2020-02-28T03:53:27.2813243Z [INFO] flink-stream-sql-test .............................. SUCCESS [  0.304 s]
2020-02-28T03:53:27.2814296Z [INFO] flink-bucketing-sink-test .......................... SUCCESS [  3.753 s]
2020-02-28T03:53:27.2815273Z [INFO] flink-distributed-cache-via-blob ................... SUCCESS [  0.270 s]
2020-02-28T03:53:27.2816627Z [INFO] flink-high-parallelism-iterations-test ............. SUCCESS [  8.881 s]
2020-02-28T03:53:27.2830226Z [INFO] flink-stream-stateful-job-upgrade-test ............. SUCCESS [  0.903 s]
2020-02-28T03:53:27.2831268Z [INFO] flink-queryable-state-test ......................... SUCCESS [  1.892 s]
2020-02-28T03:53:27.2832421Z [INFO] flink-local-recovery-and-allocation-test ........... SUCCESS [  0.139 s]
2020-02-28T03:53:27.2835929Z [INFO] flink-elasticsearch5-test .......................... SUCCESS [  3.194 s]
2020-02-28T03:53:27.2845579Z [INFO] flink-elasticsearch6-test .......................... SUCCESS [  3.591 s]
2020-02-28T03:53:27.2846591Z [INFO] flink-quickstart-test .............................. SUCCESS [  0.269 s]
2020-02-28T03:53:27.2847947Z [INFO] flink-confluent-schema-registry .................... SUCCESS [  1.587 s]
2020-02-28T03:53:27.2858556Z [INFO] flink-stream-state-ttl-test ........................ SUCCESS [  3.828 s]
2020-02-28T03:53:27.2859600Z [INFO] flink-sql-client-test .............................. SUCCESS [  1.137 s]
2020-02-28T03:53:27.2860556Z [INFO] flink-streaming-file-sink-test ..................... SUCCESS [  0.150 s]
2020-02-28T03:53:27.2861513Z [INFO] flink-state-evolution-test ......................... SUCCESS [  0.981 s]
2020-02-28T03:53:27.2872757Z [INFO] flink-rocksdb-state-memory-control-test ............ SUCCESS [  0.950 s]
2020-02-28T03:53:27.2873920Z [INFO] flink-end-to-end-tests-common ...................... SUCCESS [  1.967 s]
2020-02-28T03:53:27.2874908Z [INFO] flink-metrics-availability-test .................... SUCCESS [ 16.337 s]
2020-02-28T03:53:27.2876415Z [INFO] flink-metrics-reporter-prometheus-test ............. SUCCESS [ 17.309 s]
2020-02-28T03:53:27.2887732Z [INFO] flink-heavy-deployment-stress-test ................. SUCCESS [  7.767 s]
2020-02-28T03:53:27.2888749Z [INFO] flink-connector-gcp-pubsub-emulator-tests .......... SUCCESS [  1.046 s]
2020-02-28T03:53:27.2889897Z [INFO] flink-streaming-kafka-test-base .................... SUCCESS [  0.403 s]
2020-02-28T03:53:27.2891588Z [INFO] flink-streaming-kafka-test ......................... SUCCESS [  7.204 s]
2020-02-28T03:53:27.2898790Z [INFO] flink-streaming-kafka011-test ...................... SUCCESS [  6.475 s]
2020-02-28T03:53:27.2900043Z [INFO] flink-streaming-kafka010-test ...................... SUCCESS [  6.407 s]
2020-02-28T03:53:27.2912375Z [INFO] flink-plugins-test ................................. SUCCESS [  0.073 s]
2020-02-28T03:53:27.2913529Z [INFO] dummy-fs ........................................... SUCCESS [  0.160 s]
2020-02-28T03:53:27.2914555Z [INFO] another-dummy-fs ................................... SUCCESS [  0.181 s]
2020-02-28T03:53:27.2921875Z [INFO] flink-tpch-test .................................... SUCCESS [  0.940 s]
2020-02-28T03:53:27.2929829Z [INFO] flink-streaming-kinesis-test ....................... SUCCESS [ 13.146 s]
2020-02-28T03:53:27.2930929Z [INFO] flink-elasticsearch7-test .......................... SUCCESS [  3.704 s]
2020-02-28T03:53:27.2932805Z [INFO] flink-end-to-end-tests-common-kafka ................ SUCCESS [02:17 min]
2020-02-28T03:53:27.2944821Z [INFO] flink-tpcds-test ................................... SUCCESS [  0.310 s]
2020-02-28T03:53:27.2945931Z [INFO] flink-walkthrough-common ........................... SUCCESS [  0.945 s]
2020-02-28T03:53:27.2947182Z [INFO] flink-walkthrough-table-java ....................... SUCCESS [  3.092 s]
2020-02-28T03:53:27.2948201Z [INFO] flink-walkthrough-table-scala ...................... SUCCESS [  0.183 s]
2020-02-28T03:53:27.2959112Z [INFO] flink-walkthrough-datastream-java .................. SUCCESS [  0.191 s]
2020-02-28T03:53:27.2960510Z [INFO] flink-walkthrough-datastream-scala ................. SUCCESS [  0.168 s]
```

Reply:
`HERE` is not used to resolve the modules. It is just used for the log4j file and the artifacts file.

Reply:
I'm not sure though if the java end to end tests are properly executed, because 4 minutes seems very short.
But I wanted to investigate that another ticket

Reply:
4 minutes should actually be fine.

Reply:
Ok